### PR TITLE
Add support for INCR command

### LINKED ISSUE #2: Support command "INCR"
Let's support "[INCR](https://redis.io/commands/incr)" Redis command.

### PR DESCRIPTION
**Pull request by hedonhermdev**
Closes #2

### CODE REVIEW

**Comment by evoxmusic** [comment]
Hi @hedonhermdev, congrats on your first contribution 🥳 I am looking

File: redisless/src/server.rs [hunk 1]
```diff
@@ -184,6 +184,28 @@ fn run_command_and_get_response<T: Storage>(
...
+                            let res = format!(":{}\r\n", int_val);
+                            res.as_bytes().to_vec()
+                        } else {
+                            let res = format!("-WRONGTYPE Operation against a key holding the wrong kind of value}}");
+                            res.as_bytes().to_vec()
+                        }
+                    },
+                    None => {
+                        let val = "1";
+                        unlock(storage).set(k, val.as_bytes());
```
Comment (evoxmusic) [inline redisless/src/server.rs:204]: You can remove `let val = "1"` and do `unlock(storage).set(k, b"1");` directly

File: redisless/src/server.rs [hunk 2]
```diff
@@ -184,6 +184,28 @@ fn run_command_and_get_response<T: Storage>(
...
+                    Some(value) => {
+                        if let Ok(mut int_val) = std::str::from_utf8(value).unwrap().parse::<i64>() {
+                            int_val += 1;
+                            let new_value = int_val.to_string().into_bytes();
+                            unlock(storage).set(k.as_slice(), new_value.as_slice());
+                            let res = format!(":{}\r\n", int_val);
+                            res.as_bytes().to_vec()
+                        } else {
+                            let res = format!("-WRONGTYPE Operation against a key holding the wrong kind of value}}");
+                            res.as_bytes().to_vec()
```
Comment (evoxmusic) [inline redisless/src/server.rs:199]: you can to `b"-WRONGTYPE Operation against a key holding the wrong kind of value".to_vec()` directly

File: redisless/src/server.rs [hunk 3]
```diff
@@ -184,6 +184,28 @@ fn run_command_and_get_response<T: Storage>(
...
+            },
+            Command::Incr(k) => {
+                match unlock(storage).get(k.as_slice()) {
+                    Some(value) => {
+                        if let Ok(mut int_val) = std::str::from_utf8(value).unwrap().parse::<i64>() {
+                            int_val += 1;
+                            let new_value = int_val.to_string().into_bytes();
+                            unlock(storage).set(k.as_slice(), new_value.as_slice());
+                            let res = format!(":{}\r\n", int_val);
+                            res.as_bytes().to_vec()
```
Comment (evoxmusic) [inline redisless/src/server.rs:196]: you can return directly `format!(":{}\r\n", int_val).as_bytes().to_vec()`

File: redisless/src/server.rs [hunk 4]
```diff
@@ -184,6 +184,28 @@ fn run_command_and_get_response<T: Storage>(
...
+                        } else {
+                            let res = format!("-WRONGTYPE Operation against a key holding the wrong kind of value}}");
+                            res.as_bytes().to_vec()
+                        }
+                    },
+                    None => {
+                        let val = "1";
+                        unlock(storage).set(k, val.as_bytes());
+                        let res = format!(":{}\r\n", val);
+                        res.as_bytes().to_vec()
```
Comment (evoxmusic) [inline redisless/src/server.rs:206]: you can return directly `format!(":{}\r\n", val).as_bytes().to_vect()`

File: redisless/src/server.rs [hunk 5]
```diff
@@ -346,6 +368,11 @@ mod tests {
         let x: String = con.get("key2").unwrap();
         assert_eq!(x, "value2");
 
+        let _: () = con.set("intkey", "10").unwrap();
+        con.send_packed_command(cmd("INCR").arg("intkey").get_packed_command().as_slice()).unwrap();
```
Comment (evoxmusic) [inline redisless/src/server.rs:372]: From what I see in the Redis client documentation, there is a `con.incr(K)` method. https://docs.rs/redis/0.20.0/redis/trait.Commands.html#method.incr

**Review by evoxmusic — CHANGES_REQUESTED** [review]
Your PR is great 💪 Congrats :) It would help if you make the syntax more concise, and we are ready to merge

File: redisless/src/server.rs (hunk 5 above)
Comment (hedonhermdev) [reply redisless/src/server.rs:372]: The `con.incr(K, V)` method issues an `INCRBY` or an `INCRBYFLOAT` command. There is no `con.incr(K)` method.

**Pull request merged**